### PR TITLE
Updated xfpga userclk tests to work on HW.

### PR DESCRIPTION
**Pull request by ibarraza**
Updated the tests to enumerate based on the current platform's device id.
Separated the valid test cases.
Enabled the tests for all platforms since they will all return the same results when run in user mode.

### CODE REVIEW

**Comment by jelonanX** [comment]
Might want to fix the comment on line 206. Also, it doesn't look like we have a positive case for fpgaSetUserClock.

**Comment by ibarraza** [comment]
The comment on line 206 is accurate for that case, although the tests could be split up to make it better to understand what the test case is for. The test set_user_clock is the positive case, the issue is that for mock and dcp, this will always return FPGA_NOT_SUPPORTED, there is no case there that returns FPGA_OK. For MCP hw, it will still return FPGA_NOT_SUPPORTED unless it has root privileges, which I am still not entirely sure of the best way to handle that bit. In the case that MCP hw executes this test without root privileges, it will once again return FPGA_NOT_SUPPORTED.

File: testing/xfpga/test_usrclk_c.cpp [hunk 1]
```diff
@@ -311,5 +315,28 @@ TEST_P(usrclk_c, set_user_clock) {
...
+
+class usrclk_hw_c : public usrclk_c {};
+
+/**
+ * @test    set_user_clock
+ * @brief   Tests: xfpga_fpgaSetUserClock()
+ * @details When the parameters are valid, fpgaGetUserClock returns
+ *          FPGA_OK.
+ */
+TEST_P(usrclk_hw_c, set_user_clock) {
```
[the remark this answers is not in the document]
Comment (ibarraza) [reply testing/xfpga/test_usrclk_c.cpp]: Yep. The one get_user_clock test there will run on all platforms, mock, hw, dcp, mcp. I only had to separate the set_user_clock test case between mock and hw.

File: testing/xfpga/test_usrclk_c.cpp [hunk 2]
```diff
@@ -62,11 +62,15 @@ class usrclk_c
     system_->prepare_syfs(platform_);
 
     ASSERT_EQ(xfpga_fpgaGetProperties(nullptr, &filter_dev_), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetDeviceID(filter_dev_,
+                                        platform_.devices[0].device_id), FPGA_OK);
```
Comment (tswhison) [inline testing/xfpga/test_usrclk_c.cpp]: Nice.. I was thinking that we might need to do this in some cases.

**Comment by ibarraza** [comment]
Will retarget to master.

**Pull request closed without merging**